### PR TITLE
Initial dev server

### PR DESCRIPTION
**Pull request by timneutkens**
Move all parts that rely on `this.dev` into it's own server that is built on top of the default server.

### CODE REVIEW

**Review by HaNdTriX — APPROVED** [review]

File: server/index.js [hunk 1]
```diff
@@ -23,7 +23,6 @@ import pkg from '../../package'
 export default class Server {
   constructor ({ dir = '.', dev = false, staticMarkup = false, quiet = false, conf = null } = {}) {
     this.dir = resolve(dir)
-    this.dev = dev
     this.quiet = quiet
     this.router = new Router()
     const phase = dev ? PHASE_DEVELOPMENT_SERVER : PHASE_PRODUCTION_SERVER
```
Comment (giuseppeg) [inline server/index.js]: you could probably get rid of `dev` if you use the constructor only to define instance configuration and then add a `init` method to the class which you'd call explicitly. This way the dev server could do the extra work on its constructor

**Review by giuseppeg — APPROVED** [review]
not familiar with the codebase but the idea sounds good

File: server/index.js (hunk 1 above)
Comment (timneutkens) [reply server/index.js]: Yeah, this is just the first wave of changes, I'm going to progressively move more code to the dev server 👍

Comment (timneutkens) [reply server/index.js]: Fixed 🙌

File: server/next-dev-server.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+  }
+
+  readBuildId () {
+    return 'development'
+  }
+
+  async prepare () {
+    await super.prepare()
+    if (this.hotReloader) {
+      await this.hotReloader.start()
```
Comment (HaNdTriX) [inline server/next-dev-server.js]: We can now omit these checks.

**Pull request merged**